src/repo/repository.rs

```diff
@@ -581,7 +583,11 @@ impl Repository {
         let threads = num_cpus::get();
 
         let pack_entries = run_in_parallel(threads, files.into_iter(), |file_path| {
-            let buf = fs::read(&file_path)?;
+            let mut fd = File::open(&file_path)?;
+            let mut buf = vec![];
+            fd.read_to_end(&mut buf)?;
+            let mode = fd.metadata()?.permissions().mode();
+            drop(fd);
```

Comment:
Suggest using a scope here instead of manually dropping `fd`. Something like the following should work I think:
```rust
let (buf, mode) = {
  File::open(&file_path)?;
  // ...
  (buf, mode)
};
```

Reply:
Done in next push.